ui/src/app/store/machine/utils/common.ts

```diff
@@ -45,6 +46,22 @@ export const getMachineFieldScopes = (machine: Machine): PowerFieldScope[] => {
   return [PowerFieldScope.BMC, PowerFieldScope.NODE];
 };
 
+/**
+ * @returns Whether this machine hardware sync status should be shown.
+ */
+export function shouldShowHardwareSyncStatus(
+  machine?: Machine | null
+): machine is MachineDetails & {
+  enable_hw_sync: true;
+  status: NodeStatus.DEPLOYED;
+} {
```

Comment:
Something about using a type predicate in this util seems off, like there's a disconnect between the name of the util and what it actually does. I think if the type narrowing is necessary, the util should be renamed to reflect that (e.g. `isDeployedWithHWSync()`), otherwise this util should just return `boolean`.